### ISSUE TITLE
getProductInfo and picsCacheAll optimization 

### ISSUE THREAD

**Issue by Royalgamer06** [DoctorMcKay/node-steam-user#187]
As I see it, when `picsCacheAll` is enabled, every pics update event, the app or sub gets its product info requested and cached. Thus, cached data should always be up-to-date. So, I suggest you return cached data when `getProductInfo` is called and `picsCacheAll` is enabled.

**Comment by DoctorMcKay** [comment]
I disagree. It's not difficult to check the cache yourself if you want, and I'd rather have `getProductInfo` always return up-to-date data, since caches can get out of sync.

**Comment by Royalgamer06** [comment]
How will caches go out of sync? If pics update fails, you could delete the outdated data to prevent having outdated data.

**Comment by Royalgamer06** [comment]
Perhaps you could add a boolean to force fresh getProductInfo. Btw, remember this would only work when `picsCacheAll` is enabled, which doesn't happen that often.

**Comment by Royalgamer06** [comment]
At least, add to documentation, you could use cached `picsCache` data instead of `getProductInfo` if `picsCacheAll` is enabled.

**Comment by Royalgamer06** [comment]
I now see this:
```
				if ((!cache.apps[app.appid] && requestType == PICSRequestType.Changelist) || (cache.apps[app.appid] && cache.apps[app.appid].changenumber != data.changenumber)) {
					// Only emit the event if we previously didn't have the appinfo, or if the changenumber changed
					self.emit('appUpdate', app.appid, data);
				}
```
Why not emit if we have cache? Doesn't an updated changelist mean it's product infos has been updated? Should you update the cache as well?
